models.js

```diff
@@ -46,10 +45,37 @@ dbModel.prototype = {
   save: function(objData, options) {
     // MORETODO: returns a promise at the moment
     // mostly NOT DONE. options only has {conflict: 'update'} possibility
+    var self = this
     if (Array.isArray(objData)) {
-      return this.kninky.k.batchinsert(this.tableName, objData)
+      console.log('SAVE', objData.length, objData[0], options)
+      return this.kninky.k.batchInsert(this.tableName, objData, 100)
     } else {
+      console.log('SAVE', objData, options)
+      if (this.kninky.defaultsUnsupported) {
+        for (var a in this.fields) {
+          var f = this.fields[a]
+          if (typeof objData[a] == 'undefined'
+              && typeof f.defaultVal != 'undefined') {
+            objData[a] = f.defaultVal
+          }
+        }
+        console.log('SAVE w/defaults', objData)
+      }
+      if (options.conflict == 'update' && objData.id) {
+        //STARTHERE: look for val, if so, then send update
+        //need to turn the insert into a function which waits on result
+        return this.kninky.k.table(this.tableName)
```

Comment:
not obvious.  k is defined here:
https://github.com/MoveOnOrg/rethink-knex-adapter/blob/master/index.js#L9
It's basically the direct connection to the knex object to communicate with the sql server setup in the config.  See how/where this new adapter plugs in to the spoke app here:
https://github.com/MoveOnOrg/Spoke/blob/knex-start/src/server/models/thinky.js

So basically, in spoke, everywhere you see `r.table(....`  that's the 'direct' connection to the rethinkdb connection -- this project is re-implementing `r` (see staticR).  `k.` will be how we can run queries directly within knex paradigm (which we'll probably want to do going forward once we switch rather than force ourselves to work within the adapter framework that rethink-knex-adapter is.

In some ways, this whole project is converting `r.....` to `k.....`